add more server-types based on openssl s_client

koter84 opened the pull request:
i added the server types which are available in openssl s_client to getssl
this way you can check if the certificate has been installed correctly on more than just web and ldap servers

i will add some documentation in the next couple of days, but i wanted some feedback on the SERVER_TYPE naming.
i think for consistency it might be better to rename "webserver" to "https" (but keep "webserver" as fallback)

srvrco commented:
Thanks  :) 

naming looks fine.  I agree, change "webserver" to "https" (but keep "webserver" as fallback)

srvrco commented:
Works nicely on my tests :) 

Do you want to modify the "webserver" to "https" in this PR ? or add a new PR with updates later ? 

Is there a need to include more than one in an option ? ( for example, if you use the same cert for https, pop3, smtp ... should getssl try and check all 3 ? or just a single main one ? )

koter84 commented:
for me there is no need to use more than one server-type per certificate

also i'd think that when you are using one certificate for more than one server-type they are all linked to the same files, so if the https works, the pop3 should have also been updated.

i'll add the changes to this PR and let you know when it's ready to be pulled in

srvrco commented:
> I'd think that when you are using one certificate for more than one server-type they are all linked to the same files, so if the https works, the pop3 should have also been updated.

agreed - and makes life easier to code :) 

> i'll add the changes to this PR and let you know when it's ready to be pulled in

Thanks

koter84 commented:
i fixed the comments in the config-parts and added a table to the README to show which ports are used by which server-types.
i think i'ts ready to be pulled in

srvrco commented:
Thanks :)
